cmd/serverless/api_key_test.go

```diff
@@ -103,6 +106,23 @@ func TestExtractRegionFromMalformedPrefixSecretsManagerArnPrefix(t *testing.T) {
 	assert.Error(t, err, "could not extract region from arn: aws:secretsmanager:us-west-2:123456789012:secret:DatadogAPIKeySecret. arn: invalid prefix")
 }
 
+func TestSendAPIKeyToShellSuccess(t *testing.T) {
+	port := testutil.FreeTCPPort(t)
+	hostAndPort := fmt.Sprintf("%s%d", "localhost:", port)
+	listen, err := net.Listen("tcp", hostAndPort)
```

Comment:
I think you can easily have a race here, from what I saw `FreeTCPPort` starts listening to any available port, then closes the connection and returns the port that was used, so between this and `Listen` the port might be used again.

Reply:
I think you can use `localhost:0` for the address and then `listen.Addr().(*net.TCPAddr).Port` (*this is untested*) to get the real port of the connection.